Softeq.XToolkit.Common.Tests/Collections/ObservableItemContentRangeCollectionTests/ObservableItemContentRangeCollectionTests.cs

```diff
@@ -0,0 +1,120 @@
+            Assert.Equal(itemIndex, raisedEvent.Arguments.NewStartingIndex);
+            Assert.Equal(itemIndex, raisedEvent.Arguments.OldStartingIndex);
+
+            collection.Remove(item);
+
+            CustomAsserts.Assert_NotRaises<NotifyCollectionChangedEventArgs>(
+                   handler => collection.CollectionChanged += (s, e) => handler.Invoke(s, e),
+                   handler => { },
+                   () => item.TestProperty = TestPropertyValue2);
+        }
+
+        [Fact]
+        public void ObservableItemContentRangeCollection_WhenCreatedWithItems_NotifiesForAddedItems()
+        {
+            var item = new TestObservableObject();
+            var items = new List<TestObservableObject>() { item, new TestObservableObject(), new TestObservableObject() };
+            var collection = new ObservableItemContentRangeCollection<TestObservableObject>(items);
+            var addedItem = new TestObservableObject();
+            collection.Add(addedItem);
+
+            var itemIndex = collection.IndexOf(addedItem);
+
+            var raisedEvent = Assert.Raises<NotifyCollectionChangedEventArgs>(
+                    handler => collection.CollectionChanged += (s, e) => handler.Invoke(s, e),
+                    handler => { },
+                    () => addedItem.TestProperty = TestPropertyValue);
+
+            Assert.NotNull(raisedEvent);
+            Assert.Equal(NotifyCollectionChangedAction.Replace, raisedEvent.Arguments.Action);
+            Assert.Equal(itemIndex, raisedEvent.Arguments.NewStartingIndex);
+            Assert.Equal(itemIndex, raisedEvent.Arguments.OldStartingIndex);
+
+            collection.Remove(addedItem);
+
+            CustomAsserts.Assert_NotRaises<NotifyCollectionChangedEventArgs>(
+                   handler => collection.CollectionChanged += (s, e) => handler.Invoke(s, e),
+                   handler => { },
+                   () => addedItem.TestProperty = TestPropertyValue2);
+        }
+
+        [Fact]
+        public void ObservableItemContentRangeCollection_WhenCreatedWithoutItems_NotifiesForAddedItems()
+        {
+            var item = new TestObservableObject();
+            var collection = new ObservableItemContentRangeCollection<TestObservableObject>();
+            collection.Add(item);
+            var itemIndex = collection.IndexOf(item);
+
+            var raisedEvent = Assert.Raises<NotifyCollectionChangedEventArgs>(
+                    handler => collection.CollectionChanged += (s, e) => handler.Invoke(s, e),
+                    handler => { },
+                    () => item.TestProperty = TestPropertyValue);
+
+            Assert.NotNull(raisedEvent);
+            Assert.Equal(NotifyCollectionChangedAction.Replace, raisedEvent.Arguments.Action);
+            Assert.Equal(itemIndex, raisedEvent.Arguments.NewStartingIndex);
+            Assert.Equal(itemIndex, raisedEvent.Arguments.OldStartingIndex);
+
+            collection.Remove(item);
```

Comment:
Looks like testing two cases `Add` and `Remove` (also Remove duplicated in 3 tests), can be exported to separate method

Reply:
These test cases were reworked to include only Add or Remove logic